File: Leetcode Easy/191Numberof1Bits.py
#https://leetcode.com/problems/number-of-1-bits/submissions/1524771991/
class Solution:
    def hammingWeight(self, n: int) -> int:
        count = 0
        while n > 0:
            count += (n%2)
            n >>= 1
        return count

class Solution:
    def hammingWeight(self, n: int) -> int:
        count = 0
        while n > 0:
            count += (n%2)
            n //= 2
        return count

File: C++/Leetcode Easy/191NumberOf1Bits.cpp
class Solution {
public:
    int hammingWeight(int n) {
        int res = 0;
        while (n != 0) {
            n &= n-1;
            ++res;
        }
        return res;
    }
};

class Solution {
public:
    int hammingWeight(int n) {
        int res = 0;
        while (n != 0) {
            if (n & 1 == 1) ++res;
            n = n >> 1;
        }
        return res; 
    }
};

class Solution {
public:
    int hammingWeight(int n) {
        int res = 0;
        while (n != 0) {
            if (n & 1 == 1) ++res;
            n = n >> 1;
        }
        return res; 
    }
};